core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowFlinkPlugin.scala

```diff
@@ -25,18 +25,14 @@ import cloudflow.sbt.CloudflowKeys._
 import CloudflowBasePlugin._
 
 object CloudflowFlinkPlugin extends AutoPlugin {
-  final val FlinkVersion = "1.10.0"
-  final val CloudflowFlinkDockerBaseImage =
-    s"lightbend/flink:${CloudflowBasePlugin.CloudflowVersion}-cloudflow-flink-$FlinkVersion-scala-${CloudflowBasePlugin.ScalaVersion}"
-
   override def requires = CloudflowBasePlugin
 
   override def projectSettings = Seq(
     libraryDependencies ++= Vector(
           "com.lightbend.cloudflow" %% "cloudflow-flink"         % BuildInfo.version,
           "com.lightbend.cloudflow" %% "cloudflow-flink-testkit" % BuildInfo.version % "test"
         ),
-    cloudflowDockerParentImage := CloudflowFlinkDockerBaseImage,
+    cloudflowDockerParentImage := cloudflowDockerParentImage.value,
```

Comment:
It would be better if we can do something like:
```
cloudflowDockerParentImage := cloudflowFlinkBaseImage.value.getOrElse("s"lightbend/flink:${CloudflowBasePlugin.CloudflowVersion}-cloudflow-flink-$FlinkVersion-scala-${CloudflowBasePlugin.ScalaVersion}")
```

Reply:
where `cloudflowFlinkBaseImage` is a `settingKey[Option[String]]`. Similar for the other base images.

Reply:
Looks like having a `settingKey[Option[String]]` doesn't make the setting optional even though `getOrElse` is used. If I don't specify `cloudflowFlinkBaseImage` in `build.sbt`, then I'll get the following error:
```
[error] References to undefined settings: 
[error] 
[error]   processor / cloudflowFlinkBaseImage from processor / cloudflowDockerParentImage ((cloudflow.sbt.CloudflowFlinkPlugin.projectSettings) CloudflowFlinkPlugin.scala:39)
```
It only works if I put `cloudflowFlinkBaseImage := Option.empty[String]` in `build.sbt`, in which case the default base image will be used. Is there a way to do it without putting anything extra in `build.sbt`?

Reply:
You have to set it to None in settings in the plugin.
add `cloudflowFlinkBaseImage := None` to projectSettings of the specific plugin  (in above case CloudflowFlinkPlugin) @yuchaoran2011

Reply:
Similar stuff happens here: https://github.com/lightbend/cloudflow/blob/2cd83d8a226fb086239a1a44df10054efbdbb2a1/core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowApplicationPlugin.scala#L46

Reply:
Thanks for the suggestion. The PR has been modified based on it. I've verified that the user can now configure the base image using `cloudflowAkkaBaseImage`, `cloudflowFlinkBaseImage` or `cloudflowSparkBaseImage` in `build.sbt`. Default base images will be used in absence of user-provided setting.